### PR TITLE
Event 607 limit registrant types

### PR DESCRIPTION
**Pull request by bwtoptal**

### CODE REVIEW

File: app/scripts/directives/registrationTypeSelect.js [hunk 1]
```diff
@@ -6,8 +6,20 @@ angular.module('confRegistrationWebApp')
       templateUrl: template,
       restrict: 'E',
       controller: function ($scope, $rootScope, $location, $routeParams, RegistrationCache, uuid, modalMessage) {
+
         $scope.visibleRegistrantTypes = angular.copy($scope.conference.registrantTypes);
 
+        const findCurrentGroupRegistrantType = function() {
```
Comment (OzzieOrca) [inline app/scripts/directives/registrationTypeSelect.js]: Could you do something like this?
```js
const findCurrentGroupRegistrantType = (registrants, registrantTypes) => {
  const registrantTypeIds = registrants.map(
    ({ registrantTypeId }) => registrantTypeId
  );
  return registrantTypes.find(
    ({ allowGroupRegistrations, registrantTypeIds }) =>
      allowGroupRegistrations && registrantTypeIds.includes(type.id)
  );
};
```
I think what you've got is a little hard to read with 2 types of loops, having them nested, and the extra conditions.

Comment (bwtoptal) [reply app/scripts/directives/registrationTypeSelect.js]: I agree, thanks for the tip!

File: app/views/eventDetails/regTypes.html [hunk 2]
```diff
@@ -84,6 +83,31 @@ <h4 translate>Group Registration</h4>
                    popover-append-to-body="true">
                 </i>
               </div>
+              <div ng-if="type.familyStatus == 'ENABLED'" class="bt12">
+                <label translate class="pt12">Associated Registration Types</label>
+                <i class="question-popover fa fa-question-circle"
+                   uib-popover="{{'Select which registrant types are allowed to be added to this registration group and what is the maximum number of them' | translate}}"
```
[the remark this answers is not in the document]
Comment (bwtoptal) [reply app/views/eventDetails/regTypes.html]: Changed to what you suggested, thanks ;)

**Comment by hlbraddock** [comment]
@bwtoptal You'll want to **not** display 'Add a Group Member' on the Registration form for Group Registrations when in fact there are no more Registrant Types available to be added.

**Comment by hlbraddock** [comment]
@OzzieOrca Can you review this PR? It's close to ready.

**Comment by bwtoptal** [comment]
Thanks, @hlbraddock! Now this  'Add a Group Member' will be hidden when there are no more registrant types to be added.

File: app/views/eventDetails/regTypes.html [hunk 3]
```diff
@@ -83,6 +83,17 @@ <h4 translate>Group Registration</h4>
                    popover-append-to-body="true">
                 </i>
               </div>
+              <div class="radio">
```
Comment (hlbraddock) [inline app/views/eventDetails/regTypes.html]: @bwtoptal @OzzieOrca FYI A slight modification.

**Pull request merged**